rewrite-pattern.js

```diff
@@ -126,12 +128,20 @@ const getUnicodePropertyEscapeCharacterClassData = (property, isNegative) => {
 	return data;
 };
 
+function configNeedCaseFoldAscii() {
+	return !!config.modifiersData.i;
+}
+
+function configNeedCaseFoldUnicode() {
+	return config.modifiersData.i != false && !!((config.modifiersData.i || config.flags.ignoreCase) && config.transform.unicodeFlag);
```

Comment:
How about rewriting it into something like this for readability?

```suggestion
	if (!config.modifiersData.i) return false;
	if (!config.transform.unicodeFlag) return false;
	return Boolean(config.modifiersData.i || config.flags.ignoreCase);
```

Easier to follow IMHO

Reply:
Finish.
I used `if (config.modifiersData.i === false) return false;`, because here we need to distinguish from undefined.